src/components/Pokemon.js

```diff
@@ -0,0 +1,10 @@
+class Pokemon {
+  constructor(data) {
+    this.id = data.id;
+    this.name = data.name;
+    this.sprite = data.sprites.front_default;
+    this.type = data.types[0].type.name;
+  }
+}
+
+export default Pokemon;
```

Comment:
Maybe I mentioned it, maybe I didn't, but most of the work you will do will not use any classes. Still, it's good to know about them and how constructors and basic class inheritance work. You should be aware of basic Object Oriented Programming (OOP) concepts. So like make sure you understand how the `extends` keyword works and what `this` represents. The nice thing about classes is that it's more straightforward to add functions to it. So like let's say a Pokemon needs to "escape" or "attack". You could write class functions for these actions inside the Pokemon class. A class is nice because it's more about just holding fields of data (like an object) - it also has associated functions. But in this case here with your current code, you should also know that a class is not needed and that we can use a basic object to store those fields as well. Just be sure to keep this in mind. Some devs who come from more intensive OOP backgrounds (like Java engineers) tend to create a lot more classes for managing their data in React apps. But for those of us to haven't really used much OOP, we tend to just use plain jane objects. Both approaches are fine, but it's important to understand basic OOP in JS as well so when you do come across such code, you understand what's going on.

Reply:
Yeah, I need to study objects a little more because sometimes the "this" stuff throws me. But it's good to know that I don't have to do this all the time.